### DISCUSSION TITLE
Osqp-eigen without including eigen library. Is it possible?

### DISCUSSION THREAD

**Discussion by WestsideMS** [osqp/osqp#650] [Q&A] [answered]
I am trying to implement MPC code using osqp-eigen, but I am wondering if it is essential to include the eigen library like #include <Eigen/Dense>.

I am wondering if the code cannot be implemented using only osqp-eigen without including the eigen library.

The reason I'm asking is because I thought there might be a way to use only OSQP eigen without using the eigen library when embedding.

**Comment by traversaro** [comment, marked as answer]
Hello @WestsideMS, I am one of the maintainers of osqp-eigen . 

I am not sure what you are asking, `osqp-eigen` is a C++ binding for osqp that uses Eigen for all its types. If you do not want to use Eigen, probably you can just use osqp directly for solving your MPC problem, without using osqp-eigen? If you already have your MPC code, could you point to it so that it is more clear what you are asking?

**Reply by WestsideMS** [reply]
Hello. @traversaro,
First of all, thank you for your answer.

I understand that osqp-eigen is under the BSD license, and Eigen is under the MPL-2.0 license.
I misunderstood that by using osqp-eigen, which is under the BSD license, I could avoid Eigen's MPL-2.0 license and use the Eigen library. osqp-eigen is just a binding that allows you to conveniently use the Eigen library in C++ in osqp. I understand.
